### DISCUSSION TITLE
better compliance to nightscout REST-API

### DISCUSSION THREAD

**Discussion by floheigl** [NightscoutFoundation/xDrip#2517] [Ideas]
Hi,
with Nightscout on heroku gone, i switched a few tools/widgets etc I am using over to directly querying the xDrip on my phone.
There are a few API-discrepancies that make my life a little hard:

**Issue 1:**
A lot of ready-made pieces of software are expecting just a base-URL (mostly calling it "Nightscout URL") and then just append various paths to it expecting to find endpoints. Most prominently: 
`nightscoutURI + "/api/v1/entries/sgv.json"`
On xdrip that endpoint  is at the rootlevel (127.0.0.1:17580/sgv.json). Can we add an alias at 127.0.0.1:17580/api/v1/entries/sgv.json? 

**Issue 2:** 
Nightscout used to also include Blood glucose values in the response from sgv.json (json property "mbg"). Can we add that to xDrip 

**Issue 3:**
Can we add in the time-based filtering in the values returned ("?find[dateString][$gte]=2022-11-25T14:40:55Z")? That's really helpful to collect all the values after a given one that you already have....

I'm happy to try my hand in this, but I'd be my first contribution so i wanted to ask first.

**Comment by Navid200** [comment]
Nightscout on Heroku isn't gone.  It's just not free any longer.
There are many who are still using it.

Also, we have provided a free alternative:  https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html
xDrip is being used on the alternative as is with no issues.
There are other free alternatives as well.

Any change should leave the existing systems functional.

Usually, a question that is not answered in PRs that remain open is "why do we need this"?

We have 150 open issues.  Can you have a look and see if you can fix any of those?

I'm sorry I don't mean to be unwelcoming.  But, every change made to xDrip opens the door to more bugs.  If every developer who joins just is interested in adding more features but no interest in fixing the bugs, this will not be a sustainable process.

**Reply by floheigl** [reply]
sorry, didn't mean to question the ongoing prosperity of Nightscout, the great work done on it or the efforts to find Heroku-alternatives. I was just never really using Nightscout for anything else but the API and just realized that sending my requests directly to xdrip would simplify my setup.

What I was mainly asking for (Issue 1) is something that seems to be explicitly explained as a use-case for the pebble endpoint:
> /pebble emulates the same URL on Nightscout, so with a watchface which supports this you can enter the 
> address on the watchface, for example, Set the data endpoint on Pebble Nightscout watchface to:
> http://127.0.0.1:17580/pebble

But I have already adapted my API-clients to work with the xdrip endpoint as is, so its not an issue anymore at least for me...
